akka-projection-cassandra/src/it/scala/akka/projection/cassandra/CassandraProjectionSpec.scala

```diff
@@ -254,18 +258,14 @@ class CassandraProjectionSpec
           eventuallyExpectError(sinkProbe).getMessage should startWith(concatHandlerFail4Msg)
         }
       }
-      eventually {
-        withClue("check: projection is consumed up to third") {
-          val concatStr = repository.findById(entityId).futureValue
-          concatStr should matchPattern {
-            case Some(ConcatStr(_, "abc|def|ghi")) =>
-          }
+
+      offsetShouldBe(projectionId, 3L)
+      withClue("check: projection is consumed up to third") {
+        val concatStr = repository.findById(entityId).futureValue
+        concatStr should matchPattern {
+          case Some(ConcatStr(_, "abc|def|ghi")) =>
```

Comment:
in a few places, I swapped the checks. 

If the offset pass the check, then it's guaranteed that the concat text is also updated and we don't need another eventually around the `concatStr` checks.